api/handlers.go

```diff
@@ -251,6 +252,11 @@ func postImagesCreate(c *context, w http.ResponseWriter, r *http.Request) {
 	wf := NewWriteFlusher(w)
 
 	if image := r.Form.Get("fromImage"); image != "" { //pull
+		authConfig := dockerclient.AuthConfig{}
+		buf, err := base64.URLEncoding.DecodeString(r.Header.Get("X-Registry-Auth"))
```

Comment:
This looks good, however I was wondering why passing the auth only when using `fromImage`?

If we receive a `X-Registry-Auth` from the client, shouldn't we always pass it?